### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

**Review by tkhambir — APPROVED** [review]

File: src/main/java/com/internet/shop/dao/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Order;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/OrderDaoImpl.java]: in this PR we should only have ProductDao an ProductService

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        return Storage.products.stream()
+                .filter(productFromStorage -> productFromStorage.getId() == (product.getId()))
+                .limit(1)
+                .peek(x -> x.setPrice(product.getPrice()))
+                .peek(x -> x.setName(product.getName()))
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ProductDaoImpl.java]: what if you will ad other fields to Product?

File: src/main/java/com/internet/shop/dao/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ShoppingCartDaoImpl.java]: same

File: src/main/java/com/internet/shop/dao/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.User;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class UserDaoImpl implements UserDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/UserDaoImpl.java]: same

File: src/main/java/com/internet/shop/models/Order.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+public class Order {
+    private long id;
+    private List<Product> products;
+    private long idUser;
+
+    public Order(long id, List<Product> products, long idUser) {
+        this.id = id;
+        this.products = products;
+        this.idUser = idUser;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/models/Order.java]: let's not pass list to constructor but just initialize it here

File: src/main/java/com/internet/shop/models/ShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+public class ShoppingCart {
+    private long id;
+    private List<Product> products;
+    private long userId;
+
+    public ShoppingCart(long id, List<Product> products, long userId) {
+        this.id = id;
+        this.products = products;
+        this.userId = userId;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/models/ShoppingCart.java]: let's only initialize this list here , but not pass it to constructor

File: src/main/java/com/internet/shop/dao/OrderDaoImpl.java (hunk 1 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/dao/OrderDaoImpl.java]: deleted

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java (hunk 2 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/dao/ProductDaoImpl.java]: fixed it

File: src/main/java/com/internet/shop/dao/ShoppingCartDaoImpl.java (hunk 3 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/dao/ShoppingCartDaoImpl.java]: deleted

File: src/main/java/com/internet/shop/dao/UserDaoImpl.java (hunk 4 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/dao/UserDaoImpl.java]: deleted

File: src/main/java/com/internet/shop/models/Order.java (hunk 5 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/models/Order.java]: fixed it

File: src/main/java/com/internet/shop/models/ShoppingCart.java (hunk 6 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/models/ShoppingCart.java]: fixed it

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public Product update(Product product) {
+        return Storage.products.stream()
+                .filter(productFromStorage -> productFromStorage.getId() == (product.getId()))
+                .limit(1)
+                .peek(x -> x = product)
+                .findFirst()
+                .orElseThrow();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ProductDaoImpl.java]: does it work? will products list change after this update?

File: src/main/java/com/internet/shop/models/Order.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
+package com.internet.shop.models;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Order {
+    private long id;
+    private List<Product> products;
+    private long idUser;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/models/Order.java]: let's use Wrappers for id's

File: src/main/java/com/internet/shop/models/Order.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+    public Order(long id, long idUser) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/models/Order.java]: do you need to pass id to constuctor?

File: src/main/java/com/internet/shop/services/ProductServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.models.Product;
+import java.util.List;
+
+@Service
+public class ProductServiceImpl implements ProductServiceInterface {
+    @Inject
+    ProductDao productDao;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/services/ProductServiceImpl.java]: access modifier

File: travis.yml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (dmytro-hryhoruk) [inline travis.yml]: empty line at the end of file

Comment (frostpv) [reply travis.yml]: added

File: src/main/java/com/internet/shop/services/ProductServiceImpl.java (hunk 10 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/services/ProductServiceImpl.java]: added

File: src/main/java/com/internet/shop/models/Order.java (hunk 9 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/models/Order.java]: fixed

File: src/main/java/com/internet/shop/models/Order.java (hunk 8 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/models/Order.java]: fixed

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java (hunk 7 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/dao/ProductDaoImpl.java]: fixed

File: src/main/java/com/internet/shop/dao/ProductDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public List<Product> getAll() {
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        for (Product prod : Storage.products) {
+            if (prod.getId() == product.getId()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ProductDaoImpl.java]:
```suggestion
            if (prod.getId().equals(product.getId())) {
```
with wrappers it's better to use equals()
change it everywhere

Comment (frostpv) [reply src/main/java/com/internet/shop/dao/ProductDaoImpl.java]: fixed

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**